src/subtitles-octopus-worker.c

```diff
@@ -128,6 +145,177 @@ void libassjs_set_memory_limits(int glyph_limit, int bitmap_cache_limit) {
+                break;
+            default: {
+                    int digit = (int)(*begin - '0');
+                    if (digit < 0 || digit > 9) return 0; // invalid move
+                    value = value * 10 + digit;
+                    num_digits++;
+                    break;
+                }
+        }
+    }
+    if (num_digits > 0) {
+        params[count] = value;
+        count++;
+    }
+    if (count < 4) return 0; // invalid move
+
+    // move is animated if (x1,y1) != (x2,y2)
+    return params[0] != params[2] || params[1] != params[3];
+}
+
+static int _is_animated_tag(char *begin, char *end) {
+    // strip whitespaces around the tag
+    while (begin < end && (*begin == ' ' || *begin == '\t')) begin++;
+    while (end > begin && (end[-1] == ' ' || end[-1] == '\t')) end--;
+
+    int length = end - begin;
+    if (length < 3 || *begin != '\\') return 0; // too short to be animated or not a command
+
+    switch (begin[1]) {
+        case 'k': // fallthrough
+        case 'K':
+            // \kXX is karaoke
+            return 1;
+        case 't':
+            // \t(...) is transition
+            return length >= 4 && begin[2] == '(' && end[-1] == ')';
+        case 'm':
+            if (length >=7 && end[-1] == ')' && strcmp(begin, "\\move(") == 0) {
+                return _is_move_tag_animated(begin + 6, end - 1);
+            }
+            break;
+        case 'f':
+            // \fad() or \fade() are fades
+            return (length >= 7 && end[-1] == ')' &&
+                (strcmp(begin, "\\fad(") == 0 || strcmp(begin, "\\fade(") == 0));
+    }
+
+    return 0;
+}
+
+static int _is_event_animated(ASS_Event *event) {
+    // event is complex if it's animated in any way,
+    // either by having non-empty Effect or
+    // by having tags (enclosed in '{}' in Text)
+    if (event->Effect && event->Effect[0] != '\0') return 1;
+
+    int escaped = 0;
+    char *tagStart = NULL;
+    for (char *p = event->Text; *p != '\0'; p++) {
+        switch (*p) {
+            case '\\':
+                escaped = !escaped;
+                break;
+            case '{':
+                if (!escaped && tagStart == NULL) tagStart = p + 1;
+                break;
+            case '}':
+                if (!escaped && tagStart != NULL) {
+                    if (_is_animated_tag(tagStart, p)) return 1;
+                    tagStart = NULL;
+                }
+                break;
+            case ';':
+                if (tagStart != NULL) {
+                    if (_is_animated_tag(tagStart, p)) return 1;
+                }
+                break;
+        }
+    }
+
+    return 0;
+}
+
+static void detect_animated_events() {
+    ASS_Event *cur = track->events;
+    int *animated = is_event_animated;
+    for (int i = 0; i < track->n_events; i++, cur++, animated++) {
+        *animated = _is_event_animated(cur);
+    }
+}
+
+int libassjs_find_event_stop_times(double tm, double *eventFinish, double *emptyFinish) {
```

Comment:
So this returns current_animated which as far as I can see is essentially a bool (set using the detection function), this function name might need a tweak if so.

Reply:
You're right, though working with output arguments seems to be a bit of a pain, at least in a way it was done before. I'll fix it anyway.